### PR TITLE
Fix failing test on Windows

### PR DESCRIPTION
**Pull request by SUPERCILEX**
This test keeps failing for me on master.

### CODE REVIEW

File: subprojects/core/src/test/groovy/org/gradle/api/internal/file/BaseDirFileResolverSpec.groovy [hunk 1]
```diff
@@ -118,7 +118,7 @@ class BaseDirFileResolverSpec extends Specification {
     @Requires(TestPrecondition.WINDOWS)
     def "does not normalize windows 8.3 names"() {
         createFile(new File(tmpDir.testDirectory, 'dir/file-with-long-name.txt'))
-        def path = new File(tmpDir.testDirectory, 'dir/FILE-W~1.TXT')
+        def path = createFile(new File(tmpDir.testDirectory, 'dir/FILE-W~1.TXT'))
```
[the remark this answers is not in the document]
Comment (SUPERCILEX) [reply subprojects/core/src/test/groovy/org/gradle/api/internal/file/BaseDirFileResolverSpec.groovy]: Huh, should this test only be run on Windows 8 then?

[the remark this answers is not in the document]
Comment (oehme) [reply subprojects/core/src/test/groovy/org/gradle/api/internal/file/BaseDirFileResolverSpec.groovy]: 8.3 file name support has nothing to do with Windows 8. It's an old DOS compatibility feature.

The "assert exists" should be replaced with "assume exists". Then the test will only continue on systems that have 8.3 filename support active.

[the remark this answers is not in the document]
Comment (SUPERCILEX) [reply subprojects/core/src/test/groovy/org/gradle/api/internal/file/BaseDirFileResolverSpec.groovy]: Thanks for the help!

**Comment by SUPERCILEX** [comment]
@oehme any objections to merging?

**Comment by oehme** [comment]
Thanks @SUPERCILEX

**Pull request merged**